backend/src/Designer/Services/Implementation/OptionsService.cs

```diff
@@ -62,10 +63,33 @@ public async Task<List<Option>> GetOptionsList(string org, string repo, string d
             throw new InvalidOptionsFormatException($"One or more of the options have an invalid format in option list: {optionsListId}.");
         }
 
-
         return optionsList;
     }
 
+    /// <inheritdoc />
+    public async Task<List<RefToOptionListSpecifier>> GetAllOptionListReferences(AltinnRepoEditingContext altinnRepoEditingContext, CancellationToken cancellationToken = default)
+    {
+        cancellationToken.ThrowIfCancellationRequested();
+        AltinnAppGitRepository altinnAppGitRepository =
+            _altinnGitRepositoryFactory.GetAltinnAppGitRepository(altinnRepoEditingContext.Org,
+                altinnRepoEditingContext.Repo, altinnRepoEditingContext.Developer);
+
+        List<RefToOptionListSpecifier> optionsListReferences = new List<RefToOptionListSpecifier>();
+
+        string[] layoutSetNames = altinnAppGitRepository.GetLayoutSetNames();
+        foreach (string layoutSetName in layoutSetNames)
+        {
+            string[] layoutNames = altinnAppGitRepository.GetLayoutNames(layoutSetName);
+            foreach (var layoutName in layoutNames)
+            {
+                var layout = await altinnAppGitRepository.GetLayout(layoutSetName, layoutName, cancellationToken);
+                optionsListReferences = altinnAppGitRepository.FindOptionListReferencesInLayout(layout, optionsListReferences, layoutSetName, layoutName);
+            }
+        }
+
+        return optionsListReferences;
+    }
+
```

Comment:
Could you make a unit test for the new service method?

Reply:
Figured it was enough to test it through the endpoint, but I guess the service method theoretically can be used by other endpoints, so can add a test